app/assemblers/assemble_representation_tracks_for_select.rb

```diff
@@ -4,18 +4,38 @@ class AssembleRepresentationTracksForSelect
   initialize_with :mentor, with_feedback: Mandate::NO_DEFAULT
 
   def call
-    tracks.map { |track| SerializeTrackForSelect.(track).merge(num_submissions: track_counts[track.id]) }
+    tracks.map do |track|
+      SerializeTrackForSelect.(track).merge(num_submissions: track_num_representations[track.id])
+    end
   end
 
   private
   memoize
-  def track_counts = representations.group(:track_id).count
+  def tracks
+    Track.where(id: track_ids_with_representation).
+      where(id: track_ids_with_min_num_mentored_solutions).
+      order(title: :asc)
+  end
+
+  def track_ids_with_representation = track_num_representations.keys
+
+  def track_ids_with_min_num_mentored_solutions
+    track_num_mentored_solutions.filter_map { |track_id, count| track_id if count >= MIN_NUM_SOLUTIONS_MENTORED_PER_TRACK }
```

Comment:
I could also do this filtering in MySQL using a `HAVING` clause. Your preference?

Reply:
I'd presume HAVING is faster? Unless it somehow means some index can't be used. So happy to go with that. I don't have a big preference.